src/main/java/org/parse4j/util/ParseDecoder.java

```diff
@@ -85,6 +91,59 @@ public static Object decode(Object object) {
 		
 	}
 	
+	private static ParseObject convertJSONObjectToParseObject(JSONObject jsonObject) {
+		if (jsonObject == null) return null; 
+		ParseObject parseObject = ParseObject.createWithoutData(jsonObject.optString("className"),
+				jsonObject.optString("objectId"));
+		String[] keys = JSONObject.getNames(jsonObject);
+		if (keys != null && keys.length > 0) {
+			Map<String, Object> changeSet = new HashMap<String, Object>();
+			for (String key : keys) {
+				Object value = decode(jsonObject.opt(key));
+				if (isReservedDateType(key)) {
+					value = Parse.parseDate((String)value);
+					setMetadataThroughReflection(parseObject, key, value);
+				} else {
+					if (Parse.isValidType(value)) {
+						changeSet.put(key, value);
+					}
+				}
+				
+			}
+			if (changeSet != null && !changeSet.isEmpty()) {
+				applyChangesThroughReflection(parseObject, changeSet);
+			}
+		}
+		return parseObject;
+	}
+
```

Comment:
Here we need to add data to parseObject. Since the data field of ParseObject is protected, we have to do it through reflection. Of course we can move these code to the same pkg to avoid reflection, but this decode code is called by many other classes from other packages. we definitely don't want to mess up the package structure. change the protected fields to public is not an option either since we don't want to give users improper visibility. The reflection code will be called only twice per command, so I think it's OK to use them here. Although I haven't measure the performance diff, I do have run some commands with/without "include", and the difference I see is negligible.